marconi-chain-index/src/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -456,9 +457,22 @@ instance RI.Queryable MintBurnHandle where
         Just slotNo -> [\row -> _txMintRowSlotNo row <= slotNo]
         Nothing -> []
 
-      matchesAssetId :: C.PolicyId -> Maybe C.AssetName -> TxMintRow -> Bool
-      matchesAssetId policyId assetName row =
-        _txMintRowPolicyId row == policyId && maybe True (_txMintRowAssetName row ==) assetName
+      matchesPolicyId :: C.PolicyId -> TxMintRow -> Bool
+      matchesPolicyId policyId row = _txMintRowPolicyId row == policyId
+
+      matchesAssetName :: Maybe C.AssetName -> TxMintRow -> Bool
+      matchesAssetName assetName row = maybe True (_txMintRowAssetName row ==) assetName
+
+      matchAfterTxId :: Maybe C.TxId -> TxMintRow -> Bool
+      matchAfterTxId txid row =
+        let maybeSlot :: Maybe C.SlotNo
+            maybeSlot = txid >>= getSlotNoFromTxId
+            getSlotNoFromTxId :: C.TxId -> Maybe C.SlotNo
+            getSlotNoFromTxId txid' =
+              if row ^. txMintRowTxId == txid'
+                then Just $ row ^. txMintRowSlotNo
+                else Nothing
+         in maybe True (row ^. txMintRowSlotNo >=) maybeSlot
```

Comment:
It still doesn't work: we need to resolve the given `TxId` against every stored txs. Either I read it wrong or we're just comparing it to the current row.

Reply:
we do that for the database part, [see a few lines below](https://github.com/input-output-hk/marconi/pull/86/files#diff-2be21c02626d215366ba0a9977ff0f3d871dbc7828bbd60d4f0e8bd15baeb929R517), isn't that sufficient since when we query, we query both in-memory and SQLite?

Reply:
It isn't, if the provided txid is in memory. And if it was, then this second check still wouldn't help: aren't we comparing the slot number of a txid with itself here?

Reply:
rrrrrr, yes, the logic is incorrect and useless, thnx for the analysis.
For the in-memory events,  it should be sufficient to 
- find the txId's corresponding slotNo from entire in-memory portion
- add that slotNo to the `afterTx` rowFilter
Yes?
I'll make the changes ASAP

Reply:
Yes, and if you find the slot in the in-memory part, you can just skip the database result.